### PR TITLE
Bind internal pinning observer lifetime to internal arena

### PR DESCRIPTION
**Pull request by ivankochin**
Now the internal `numa_binding_observer` inside the arena destroys when the `tbb::task_arena` reference destroys. But enqueued tasks may not still be executed, so we should prolong the observer lifetime to the internal arena lifetime.

Signed-off-by: Kochin Ivan <kochin.ivan@intel.com>

### CODE REVIEW

**Comment by ivankochin** [comment]
This MR requires the SpinBarrier lifetime udate from [this merge request](https://github.com/oneapi-src/oneTBB/pull/482) to work correctly.

File: src/tbb/arena.cpp [hunk 1]
```diff
@@ -231,6 +231,12 @@ void arena::free_arena () {
 #if __TBB_ENQUEUE_ENFORCED_CONCURRENCY
     __TBB_ASSERT( !my_global_concurrency_mode, NULL );
 #endif
+#if __TBB_ARENA_BINDING
+    if(my_numa_binding_observer != nullptr ) {
```
Comment (pavelkumbrasev) [inline src/tbb/arena.cpp:235]:
```suggestion
    if (my_numa_binding_observer != nullptr) {
```

File: src/tbb/governor.cpp [hunk 2]
```diff
@@ -51,6 +51,10 @@ namespace rml {
 tbb_server* make_private_server( tbb_client& client );
 } // namespace rml
 
+namespace system_topology {
+    void destroy();
```
Comment (pavelkumbrasev) [inline src/tbb/governor.cpp:55]: In governor you use system in tbb_bind platform, so please consider use one of them.
```suggestion
    void destroy_platform_topology();
```

File: test/common/common_arena_constraints.h [hunk 3]
```diff
@@ -452,13 +452,12 @@ system_info::affinity_mask get_arena_affinity(tbb::task_arena& ta) {
         arena_affinity = system_info::allocate_current_affinity_mask();
     });
 
-    utils::SpinBarrier barrier;
-    barrier.initialize(ta.max_concurrency() - 1);
+    utils::SpinBarrier exit_barrier;
+    exit_barrier.initialize(ta.max_concurrency());
```
Comment (pavelkumbrasev) [inline test/common/common_arena_constraints.h:456]:
```suggestion
    utils::SpinBarrier exit_barrier(ta.max_concurrency());
```

**Review by pavelkumbrasev — COMMENTED** [review]
Consider use one of 'platform' or 'system'.

**Review by pavelkumbrasev — APPROVED** [review]

**Pull request merged**